src/modules/sensor.c

```diff
@@ -0,0 +1,70 @@
+#include <avr/io.h>
+#include "sensor.h"
+#include "../control.h"
+#include "../connector.h"
+#include "light_sensor.h"
+#include "temp_sensor.h"
+
+volatile int status = 1;
+
+float read_sensor()
+{
+	if (SENSOR_MODE == 0)
```

Comment:
Can be simplified to ``return SENSOR_MODE == 0 ? read_light() : read_temp();``, since there are only two possible values.